### PR TITLE
Issue516 docs refactor

### LINKED ISSUE #138: Update trademarks in docs to match F5 branding style guide
Per updated style guide (June 2016): "Apply trademarks to first mentions of products and technologies in body copy (rather than headlines, subheads, captions, etc.)." 

Ongoing -- update docs as they're edited.

### PR DESCRIPTION
**Pull request by jputrino**
@rsalsa 
@pjbreaux 

#### What issues does this address?
Fixes #138 #517 #523 #669

#### What's this change do?
Refactors the F5 LBaaSv2 driver documentation into "product" docs.
Sets up deployment to the s3 bucket for clouddocs.f5.com.

#### Where should the reviewer start?
You can view the staged docs via the link shared in Slack.

### CODE REVIEW

**Comment by pjbreaux** [comment]
@jputrino: Sorry for the broken build, I introduced build on PR and it kicked these builds. We can ignore the failing build with my name on it. Trust travis for now.

**Comment by pjbreaux** [comment]
Can one of the admins verify this patch?

**Comment by rsalsa** [comment]
@jputrino - is the sidebar supposed to float on the right had side?

**Comment by jputrino** [comment]
@rsalsa -- this is what it is supposed to look like:
https://s3-us-west-2.amazonaws.com/staging-c2ub89n2qjgt1/products/openstack/lbaasv2-driver/index.html

File: requirements.os.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-e .
```
Comment (pjbreaux) [inline requirements.os.txt]: Where is this file used? I see no references here to it.

Comment (pjbreaux) [inline requirements.os.txt]: I'm referring to requirements.os.txt above

**Comment by pjbreaux** [comment]
@jputrino: I cancelled the smoke build, as I am currently testing it on mitaka. I will kick it once that is done.

File: requirements.os.txt (hunk 1 above)
Comment (jputrino) [inline requirements.os.txt]: Good catch! That doesn't need to be here.

File: .travis.yml [hunk 2]
```diff
@@ -60,4 +62,22 @@ deploy:
...
       python: 2.7
     skip_cleanup: true
-
+  # deploy docs to s3
+  - provider: script
+    skip_cleanup: true
+    on:
+      repo: F5Networks/f5-openstack-lbaasv2-driver
+      condition: $TRAVIS_TAG = ""
+    # if this is not a tagged release and we're on master, the docs go to /products/openstack/lbaasv2-driver/latest
```
Comment (pjbreaux) [inline .travis.yml]: Is the branch included in this url? This is unclear.

**Comment by pjbreaux** [comment]
retest this please

**Pull request merged**